### DISCUSSION TITLE
Guide to Soroban + Rust – building

### DISCUSSION THREAD

**Discussion by wildework** [stellar/sorobanathon#69] [General]
This is my third guide submission, this time I tackle the Soroban contract building process.

Coming from a JavaScript developer background, the building/compilation step is not super familiar, so I dig deeper into the differences between JavaScript and Rust, as well as what's going on under the hood. I even dig into the compiled WebAssembly Soroban contract binary and explain how the Rust code maps to it.

[sorobandev.com/guides/building](https://www.sorobandev.com/guides/building)

**Comment by achallagundla** [comment]
Heya @wildework ,
Congratulations! Your contribution to Sorobanathon: First Light has been accepted! 🎉
Your submission may qualify for a reward of 1000 XLM! One last step: to receive that reward, please fill out this [form](https://stellarform.typeform.com/Sorobanathon) completely and accurately.

Please let us know if you have any questions. (You can also message me on discord @ anuxhya)

All the best,
Anuhya

PS: No need to submit the form if you already have, just update me on discord if your wallet address has changed.

**Reply by wildework** [reply]
Hi @achallagundla thank you for your comment, I just wanted to clarify - are you awarding ElliotFriend for my post? Thanks!

**Reply by achallagundla** [reply]
Oops that was a typo on my end
